packages/client/src/components/ReviewDialog/ReviewDialog.jsx

```diff
@@ -46,46 +72,122 @@ export function ReviewDialog({ initialState, handleClose, user, movieId }) {
+            defaultValue={formData.review_text}
+            onChange={(e) => {
+              setFormData({ ...formData, review_text: e.target.value });
+            }}
+          />
+        ) : (
+          <ReviewContentText
+            autoFocus
+            margin="dense"
+            id="name"
+            label="review"
+            multiline
+            rows={4}
+            fullWidth
+            variant="filled"
+            InputLabelProps={{
+              sx: {
+                color: 'mainGreen',
+                textTransform: 'capitalize',
+              },
+            }}
+            placeholder="Leave your first review about this movie"
+            InputProps={{ sx: { color: 'mainGreen' } }}
+            onChange={(e) => {
+              setFormData({ ...formData, review_text: e.target.value });
+            }}
+          />
+        )}
       </ReviewContent>
       <DialogActions
-        sx={{ backgroundColor: 'backgroundDark', color: 'mainGreen' }}
+        sx={{
+          backgroundColor: 'backgroundDark',
+          color: 'mainGreen',
+          display: 'flex',
+        }}
       >
-        <Button onClick={handleClose} sx={{ backgroundColor: 'mainGreen' }}>
-          Delete review
-        </Button>
-        <Button onClick={handleClose} sx={{ backgroundColor: 'mainGreen' }}>
-          Save review
-        </Button>
-        <Button onClick={handleClose} sx={{ backgroundColor: 'mainGreen' }}>
-          Update review
-        </Button>
+        {review ? (
+          <>
+            <Button
+              onClick={() => {
+                deleteReview(review.reviewID);
+                handleClose();
+              }}
+              sx={{ backgroundColor: 'mainGreen' }}
+            >
+              Delete review
+            </Button>
+            <Button
+              onClick={() => {
+                updateReview(review.reviewID);
+                handleClose();
+              }}
+              sx={{ backgroundColor: 'mainGreen' }}
+            >
+              Update review
+            </Button>
+            <Button
+              onClick={handleClose}
+              sx={{ backgroundColor: 'mainGreen' }}
+              disabled
+            >
+              Save review
+            </Button>
+          </>
+        ) : (
+          <>
+            <Button
+              onClick={handleClose}
+              sx={{ backgroundColor: 'mainGreen' }}
+              disabled
+            >
+              Delete review
+            </Button>
+            <Button
+              onClick={handleClose}
+              sx={{ backgroundColor: 'mainGreen' }}
+              disabled
+            >
+              Update review
+            </Button>
+            <Button
+              onClick={() => {
+                postReview();
+                handleClose();
+              }}
+              sx={{ backgroundColor: 'mainGreen' }}
+            >
+              Save review
+            </Button>
+          </>
+        )}
```

Comment:
This part of code is rather long and 90% of code just repeats previous part. All the buttons and text area inside ? ternary operator have the same properties except for a few. As far as I can see, only disabled property changes...
Maybe it would be better to put condition only to this property and not repeat the code for the buttons themselves?
something like disabled={review ? false : true}

Reply:
Okey!It's a ggod idea.I will change